otter/util/zk.py

```diff
@@ -170,3 +199,113 @@ def get_zk_dispatcher(kz_client):
             partial(perform_get_stat, kz_client),
         AcquireLock: perform_acquire_lock
     })
+
+
+class PollingLock(object):
+    """
+    Zookeeper lock recipe that polls the children on interval basis instead
+    of leaving a watch on previous child. It's supposed to replace
+    :obj:`kazoo.recipe.lock.Lock` and has same signature and _similar_ behavior
+    for ``acquire`` and ``release`` methods.
+    """
+
+    def __init__(self, dispatcher, path, identifier=""):
+        self.dispatcher = dispatcher
+        self.path = path
+        self.identifier = identifier
+        self._node = None
```

Comment:
> Hmm. I don't know how that pattern will help here. Moreover, it looks you want to use it because you want to use it :) ? Did you have any reasons for this?

This talk is a good motivation for pretty much *always* using this pattern: https://www.youtube.com/watch?v=29MAL8pJImQ

If this code is being changed in the future, with a `None`-or-not-`None` variable, you have to be *careful* about how you touch it.  You have to account for both cases at every call site.  You have to carefully consider the significance of `None` and re-derive it if you're not quite sure.  (I had to think pretty hard about what states `None` reflected, since it actually reflects 2 out of 3.)

With an Active Nothing, instead, you can blindly trust that you can send the same set of messages to the same attribute at any time.  It might *do something different*, but you don't have to be *careful*.  This is why Automat is a possible alternative to Active Nothing - with Automat you can "blindly" touch variables because the state-machine constraints will prevent you from entering a state where it isn't set.  It's still not quite as good for this sort of thing, but it also lets you deal with more complex state hierarchies where the "messages being responded to" is less clear (and I do think it's less clear here, or I'd just suggest an interface).

In any case - this is just something I will bring up *every* time I see a `None` now, because `None` is Python's manifestation of [the billion dollar mistake](https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare) and one day I'd like to stop making it.